### PR TITLE
ci: fix unauthed deploys

### LINKED ISSUE #339: Pushing to new branch without PR causes deployment CI to run
### Describe the bug

We need to check if we are on main in checks like the following:
https://github.com/helpwave/services/blob/main/.github/workflows/cicd.yaml#L102

### Describe how to produce the bug

1. Create a new branch a push changes to it
2. Don't create a PR for it
3. Change is deployed

### Additional context

https://github.com/helpwave/services/actions/runs/5762996167/job/15623902760

### PR DESCRIPTION
**Pull request by FoseFx**
## Which issues does this pull request close?
closes #339

### CODE REVIEW

**Comment by FoseFx** [comment]
I know what you're thinking, and I agree. I would also love to store this in a variable for the whole workflow, but, afaik, you can only define environment variables, not github(?) variables, for a workflow. But a job is able to overwrite this. If a malicious actor writes go code to modify the environment, they might be able to force a deployment.

**Comment by MaxSchaefer** [comment]
@FoseFx any news?

**Comment by FoseFx** [comment]
I removed the env again and now it works. I'd leave it as it is, if you want to open a follow-up issue to improve it, feel free to do so.

File: .github/workflows/buf.yaml [hunk 1]
```diff
@@ -42,6 +42,8 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: bufbuild/buf-setup-action@v1
+        with:
+          github_token: ${{ github.token }}
```
Comment (MaxSchaefer) [inline .github/workflows/buf.yaml:46]: Reason?

**Review by MaxSchaefer — APPROVED** [review]

File: .github/workflows/buf.yaml (hunk 1 above)
Comment (FoseFx) [reply .github/workflows/buf.yaml:46]: The wf was flaky due to rate-limiting (unauthed requests)

**Pull request merged**